velox/common/memory/Memory.h

```diff
@@ -577,17 +394,14 @@ class IMemoryManager {
   // that would cause a quota breach results in exceptions.
   virtual int64_t getMemoryQuota() const = 0;
   // Power users that want to explicitly modify the tree should get the root of
-  // the tree. Users opting to use this api needs to explicitly manage the life
-  // cycle of the memory pools by calling removeSelf() after use.
+  // the tree.
   // TODO: perhaps the root pool should be a specialized pool that
   //        * doesn't do allocation
   //        * cannot be removed
   virtual MemoryPool& getRoot() const = 0;
 
-  // Adds a child pool to root for use. The actual return type is
-  // std::unique_ptr of a wrapper class that knows to remove itself.
-  virtual std::unique_ptr<ScopedMemoryPool> getScopedPool(
-      int64_t cap = kMaxMemory) = 0;
+  // Adds a child pool to root for use.
+  virtual std::shared_ptr<MemoryPool> getPool(int64_t cap = kMaxMemory) = 0;
```

Comment:
getPool -> addChild?

Would be nice to document the ownership model, e.g. clarify that this class does not own the new child.